### PR TITLE
Shorten languages select list & connect chatbox & editor with backend

### PR DESCRIPTION
**Pull request by LinkCut**
- Shorten languages list to support popular languages
- Connect chatbox with backend service
- Support code submission to ML side
- Display AI response in chatbox from ML side

(Use `Squash and merge` to have a better look of commit history at master branch)

### CODE REVIEW

File: src/contexts/MessagesContext.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  dispatch: Dispatch<MessageActions>;
+}>({
+  messages: initialMessages,
+  dispatch: () => null
+});
+
+MessagesContext.displayName = "MessagesContext";
+
+export function InterviewContextProvider({children}: React.PropsWithChildren<{}>) {
+  // Use the useReducer hook to manage the cart state
```
Comment (tan-i-ham) [inline src/contexts/MessagesContext.tsx]: what is cart state? maybe interview state?

**Review by tan-i-ham — APPROVED** [review]
LGTM

File: src/contexts/MessagesContext.tsx (hunk 1 above)
Comment (LinkCut) [reply src/contexts/MessagesContext.tsx]: Oops, I forgot to clean up some useless comments that were copied from the previous project. Fixed. Feel free to take a look at the latest changes.

**Review by tan-i-ham — APPROVED** [review]

**Comment by zhFIsHappy** [comment]
Already merged

**Pull request merged**